internal/batches/docker/image_test.go

```diff
@@ -76,6 +79,26 @@ func TestImage_Digest(t *testing.T) {
 			}
 		})
 	}
+
+	t.Run("docker timeout", func(t *testing.T) {
+		// We're more interested in the context error handling than the parsing
+		// of the timeout, and we don't want to slow down the test, so we're
+		// going to construct a context that has already exceeded its deadline
+		// at the point it is provided to Digest.
+		ctx, cancel := context.WithTimeout(context.Background(), -1*time.Second)
```

Comment:
Lol, didn't know that's valid!

Reply:
[Turns out I didn't even invent it!](https://sourcegraph.com/search?q=context:global+lang:go+context.WithTimeout%28...%2C+-...%29&patternType=structural) I particularly like that it's used in the runtime's own unit tests.

Reply:
Nice, so they would break their own tests before it ships in a go release 😆